### PR TITLE
Implement TestRun running state check before deletion

### PR DESCRIPTION
**Pull request by kforeverisback**
# Type of PR

- [X] Documentation changes
- [X] Code changes
- [X] Test changes

## Purpose of PR

- Include CompletedTime in TestRun and TestRunPayload object
- Check existence of CompletedTime before deletion
- Return 409 if TestRun is still running
- Add Integration test
- Update SchemaFilters

Below shows the table where two operations in Delete method is performed and the return status based on the status of those two operations:

| Get | Delete | Returns | Message |
| :--- | :------- | :-------- | :--------- |
IntErr    | Any       | IntErr    | Cosmos error
NotFound  | Any       | NotFound  | ID not found*
Ok        | NotFound  | NotFound  | (Redundant) Same as above*
Ok        | Conflict  | Conflict  | ID Found but unfinished
Ok        | Ok        | NoContent | All good
Any       | Any       | IntErr    | But also show GET and DELETE status           

## Does this introduce a breaking change

- [ ] YES
- [X] NO

## Validation

- [X] Unit tests updated and ran successfully
- [X] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes https://github.com/retaildevcrews/wcnp/issues/584

### CODE REVIEW

File: src/LodeRunner.API/src/Controllers/TestRunsController.cs [hunk 1]
```diff
@@ -132,6 +133,10 @@ public async Task<ActionResult> CreateTestRunConfig([FromBody, SwaggerRequestBod
                 return await ResultHandler.CreateCancellationInProgressResult();
             }
 
+            // Explicitly set CompletedTime to null, in case CompletedTime is set in TestRunPayload
+            // TODO: Would have to create a new object to make AutoMapper ignore CompletedTime for Post only. Can we do better? maybe even using [Bind]
```
Comment (kforeverisback) [inline src/LodeRunner.API/src/Controllers/TestRunsController.cs:137]: @gortegaMS @joheec
This slightly feels like a "small" hack.
Hence the TODO: Comment.

File: src/LodeRunner.API/src/Controllers/TestRunsController.cs [hunk 2]
```diff
@@ -160,38 +165,65 @@ public async Task<ActionResult> CreateTestRunConfig([FromBody, SwaggerRequestBod
         /// <returns>IActionResult.</returns>
         [HttpDelete("{testRunId}")]
         [SwaggerResponse((int)HttpStatusCode.NoContent, SystemConstants.DeletedTestRun)]
-        [SwaggerResponse((int)HttpStatusCode.BadRequest, SystemConstants.InvalidTestRunId, typeof(Middleware.Validation.ValidationError), "application/problem+json")]
         [SwaggerResponse((int)HttpStatusCode.NotFound, SystemConstants.NotFoundTestRun)]
         [SwaggerResponse((int)HttpStatusCode.ServiceUnavailable, SystemConstants.TerminationDescription)]
         [SwaggerResponse((int)HttpStatusCode.InternalServerError, SystemConstants.UnableToDeleteTestRun)]
+        [SwaggerResponse((int)HttpStatusCode.InternalServerError, SystemConstants.Unknown)]
```
Comment (kforeverisback) [inline src/LodeRunner.API/src/Controllers/TestRunsController.cs:171]: @gortegaMS Is it possible to add Two message for same StatusCode in Swagger?

File: src/LodeRunner.API/src/Controllers/TestRunsController.cs [hunk 3]
```diff
@@ -160,38 +165,65 @@ public async Task<ActionResult> CreateTestRunConfig([FromBody, SwaggerRequestBod
...
 
-            var deleteTaskResult = await testRunService.Delete(testRunId);
-
-            return deleteTaskResult switch
+            return (existingTestRunResp.StatusCode, delStatusCode) switch
             {
-                HttpStatusCode.OK => await ResultHandler.CreateNoContent(),
-                HttpStatusCode.NotFound => await ResultHandler.CreateErrorResult(SystemConstants.NotFoundTestRun, HttpStatusCode.NotFound),
-                _ => await ResultHandler.CreateErrorResult(SystemConstants.UnableToDeleteTestRun, HttpStatusCode.InternalServerError),
+                (HttpStatusCode.InternalServerError, _) =>
```
Comment (kforeverisback) [inline src/LodeRunner.API/src/Controllers/TestRunsController.cs:201]: Refer to the Table in the PR description

**Review by jofultz — APPROVED** [review]

**Review by gortegaMS — CHANGES_REQUESTED** [review]
Some of the checks failed please take a look

File: src/LodeRunner.API/src/Controllers/TestRunsController.cs (hunk 2 above)
Comment (gortegaMS) [reply src/LodeRunner.API/src/Controllers/TestRunsController.cs:167]: I am not sure, but I do not think so

**Review by gortegaMS — APPROVED** [review]

**Comment by kforeverisback** [comment]
CodeQL build failed at first. Rerunning CodeQL action succeeded.

**Pull request merged**